cdap-app-templates/cdap-etl/cdap-etl-batch-app/src/main/java/co/cask/cdap/app/etl/batch/EmailAction.java

```diff
@@ -0,0 +1,143 @@
+ */
+public class EmailAction extends AbstractWorkflowAction {
+  private static final String NAME = "Email";
+  private static final String DEFAULT_HOST = "localhost";
+  private static final int DEFAULT_PORT = 25;
+  private static final String DEFAULT_PROTOCOL = "smtp";
+
+  public static final String RECIPIENT_EMAIL_ADDRESS = "recipientEmailAddress";
+  public static final String FROM_ADDRESS = "senderEmailAddress";
+  public static final String MESSAGE = "message";
+  public static final String SUBJECT = "subject";
+  public static final String USERNAME = "username";
+  public static final String PASSWORD = "password";
+  public static final String PROTOCOL = "protocol";
+  public static final String HOST = "host";
+  public static final String PORT = "port";
+
+  private Map<String, String> properties;
+
+  public EmailAction(ETLStage action) {
+    super(action.getName());
+    properties = action.getProperties();
+  }
+
+  public WorkflowActionSpecification configure() {
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(properties.get(RECIPIENT_EMAIL_ADDRESS)),
+                                String.format("You must set the \'%s\' property to send an email.",
+                                              RECIPIENT_EMAIL_ADDRESS));
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(properties.get(FROM_ADDRESS)),
+                                String.format("You must set the \'%s\' property to send an email.", FROM_ADDRESS));
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(properties.get(MESSAGE)),
+                                String.format("You must set the \'%s\' property to send an email.", MESSAGE));
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(properties.get(SUBJECT)),
+                                String.format("You must set the \'%s\' property to send an email.", SUBJECT));
+    Preconditions.checkArgument(!(Strings.isNullOrEmpty(properties.get(USERNAME)) ^
+                                  Strings.isNullOrEmpty(properties.get(PASSWORD))),
+                                String.format("You must either set both username and password " +
+                                                "or neither username nor password. " +
+                                                "Currently, they are username: \'%s\', and password: \'%s\'",
+                                              properties.get(USERNAME), properties.get(PASSWORD)));
+    if (Strings.isNullOrEmpty(properties.get(HOST))) {
+      properties.put(HOST, DEFAULT_HOST);
+    }
+    if (Strings.isNullOrEmpty(properties.get(PORT))) {
+      properties.put(PORT, Integer.toString(DEFAULT_PORT));
+    }
+    return WorkflowActionSpecification.Builder.with().setName(NAME).setDescription("")
+      .withOptions(properties).build();
+  }
+
+  @Override
+  public void run() {
+    properties = ((WorkflowActionNode) getContext().getWorkflowSpecification().getNodeIdMap()
+      .get(NAME)).getActionSpecification().getProperties();
```

Comment:
The properties is static and won't change in runtime time. Why not do the java mail properties setting in initialize time?